### PR TITLE
o/ifacestate: special-case system-files and force refreshing its static attributes

### PR DESCRIPTION
**Pull request by stolowski**
Fixes LP: #1942266 

(for more context - previous fix for content interface is here https://github.com/snapcore/snapd/pull/6802)

It includes the spread test originally proposed by @anonymouse64 , with some tweaks.

### CODE REVIEW

File: tests/regression/lp-1942266/task.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+prepare: |
+  # build the first version
+  sed -e "s/##VERSION##/0.1/" ./test-system-files-conn-snap/meta/snap.yaml.in > ./test-system-files-conn-snap/meta/snap.yaml
+  sed -i -e "s/##EXTRA##//" ./test-system-files-conn-snap/meta/snap.yaml
+  snap pack ./test-system-files-conn-snap --filename=test-system-files-conn-snap-1.snap
+
+  # build the second version
+  sed -e "s/##VERSION##/0.2/" ./test-system-files-conn-snap/meta/snap.yaml.in > ./test-system-files-conn-snap/meta/snap.yaml
+  sed -i -e 's@##EXTRA##@- /etc/foo.conf @' ./test-system-files-conn-snap/meta/snap.yaml
+
```
Comment (anonymouse64) [inline tests/regression/lp-1942266/task.yaml:12]:
```suggestion
```

File: tests/regression/lp-1942266/task.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  snap pack ./test-system-files-conn-snap --filename=test-system-files-conn-snap-2.snap
+
+execute: |
+  echo "With the first revision, we don't have the rule because the interface is not connected"
+  snap install --dangerous test-system-files-conn-snap-1.snap
```
Comment (anonymouse64) [inline tests/regression/lp-1942266/task.yaml:17]: seems I forgot to add this but we should actually check that the rule isn't there to be double extra sure

```suggestion
  snap install --dangerous test-system-files-conn-snap-1.snap
  NOMATCH /etc/resolv.conf < /var/lib/snapd/apparmor/profiles/snap.test-system-files-conn-snap.bin
  NOMATCH /etc/foo.conf < /var/lib/snapd/apparmor/profiles/snap.test-system-files-conn-snap.bin
```

**Review by anonymouse64 — APPROVED** [review]
LGTM, thanks for picking this up and fixing my silly mistakes in the spread test :smile: 

One nitpick (about my spread test)

**Review by mvo5 — APPROVED** [review]
Looks good, thank you!

**Comment by alexmurray** [comment]
I wonder if perhaps it was intentional that changes to system-files would not get automatically updated since this does grant additional access to the application which the user may not expect - however, since system-files changes need to have an accompanying snap declaration update from the store then changes to get reviewed by the store team and have to be granted there. As such I don't think there is a problem with making this change (and I think it is a good usability enhancement too).

**Comment by alexmurray** [comment]
Do we need similar logic for `personal-files` as well?

**Comment by anonymouse64** [comment]
@alexmurray for now, just system-files since that's what our customer(s) need, I think for personal-files we can wait until someone urgently needs it or until we have time to do the full thing

**Comment by stolowski** [comment]
> I wonder if perhaps it was intentional that changes to system-files would not get automatically updated since this does grant additional access to the application which the user may not expect - however, since system-files changes need to have an accompanying snap declaration update from the store then changes to get reviewed by the store team and have to be granted there. As such I don't think there is a problem with making this change (and I think it is a good usability enhancement too).

Granting additional access which the user may not expect is valid concern I think, but current behavior was never intentional and it's a long standing bug that we don't refresh interface attributes when snaps get refreshed (and it affects all interfaces that have attributes).

**Comment by anonymouse64** [comment]
I had to push a super simple fix to the spread test so it only runs on systems with AppArmor confinement, after this batch of tests run, this is good for 2.52

**Pull request merged**